hudi-common/src/main/java/org/apache/hudi/metadata/FileSystemBackedTableMetadata.java

```diff
@@ -105,6 +106,20 @@ public FileSystemBackedTableMetadata(HoodieEngineContext engineContext, Serializ
     return partitionPaths;
   }
 
+  @Override
+  public Map<String, FileStatus[]> getAllFilesInPartitions(List<String> partitionPaths)
+      throws IOException {
+    int parallelism = Math.min(DEFAULT_LISTING_PARALLELISM, partitionPaths.size());
```

Comment:
If the `partitionPaths` is empty, the `parallelism` will be 0, there may be an Exception ("Positive number of partitions required")  throw out for the `sparkContext.parallelize(seq, parallelism)`,

Reply:
Will add a check to throw a more graceful exception. There should ideally be atleast one partition path. Even in case of non-partitioned tables, it expects a path.

Reply:
Hi @umehrot2 , I think we should return an empty FileStatus array if the partition paths is empty. Return an exception will affect  query on hudi table. Some people has report this issue at https://github.com/apache/hudi/issues/2935

Reply:
Makes sense. Done.